src/storage/mod.rs

```diff
@@ -1585,6 +1586,57 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                 .await?
         }
     }
+
+    pub fn raw_compare_and_set_atomic(
+        &self,
+        ctx: Context,
+        cf: String,
+        key: Vec<u8>,
+        previous_value: Vec<u8>,
+        value: Vec<u8>,
+        cb: Callback<Option<Value>>,
+    ) -> Result<()> {
+        let cf = Self::rawkv_cf(&cf)?;
+        let previous = if previous_value.is_empty() {
+            None
+        } else {
+            Some(previous_value)
+        };
+        let cmd = AtomicCompareAndSet::new(cf, Key::from_encoded(key), previous, value, ctx);
+        self.sched_txn_command(cmd, cb)
+    }
+
+    pub fn raw_batch_put_atomic(
```

Comment:
Why do we need it? Isn't a normal write batch atomic?

Reply:
Emmmm, it is just for `batch_put`.....
If we do not supply such a method, user can not change data rightly.

Reply:
ok.. I understand it from the updated comment.